samples/snippets/data_client/data_client_snippets_async.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+
+# Copyright 2023, Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+async def write_simple(table):
+    # [START bigtable_async_write_simple]
+    from google.cloud.bigtable.data import BigtableDataClientAsync
+    from google.cloud.bigtable.data import SetCell
+
+    async def write_simple(project_id, instance_id, table_id):
+        async with BigtableDataClientAsync(project=project_id) as client:
+            async with client.get_table(instance_id, table_id) as table:
+                family_id = "stats_summary"
+                row_key = b"phone#4c410523#20190501"
+
+                cell_mutation = SetCell(family_id, "connected_cell", 1)
+                wifi_mutation = SetCell(family_id, "connected_wifi", 1)
+                os_mutation = SetCell(family_id, "os_build", "PQ2A.190405.003")
+
+                await table.mutate_row(row_key, cell_mutation)
+                await table.mutate_row(row_key, wifi_mutation)
+                await table.mutate_row(row_key, os_mutation)
+
+    # [END bigtable_async_write_simple]
+    await write_simple(table.client.project, table.instance_id, table.table_id)
+
+
+async def write_batch(table):
+    # [START bigtable_async_writes_batch]
+    from google.cloud.bigtable.data import BigtableDataClientAsync
+    from google.cloud.bigtable.data.mutations import SetCell
+    from google.cloud.bigtable.data.mutations import RowMutationEntry
+
+    async def write_batch(project_id, instance_id, table_id):
```

Comment:
I think we should use batcher for the write_batch example

Reply:
Should I change the non-async samples to be consistent?

Reply:
If you have time to do that then that'll be great! Otherwise I think it can be a fast follow

Reply:
sure, added for both!